.github/workflows/test_python.yml

```diff
@@ -0,0 +1,29 @@
+name: test nodl_python
+
+on:
+  pull_request:
+    paths:
+      - '**.py'
+  push:
+    branches:
+      - master
+  schedule:
+    - cron: '0 * * * *'
+
+jobs:
+  build-and-test:
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os: [ubuntu-latest, windows-latest, macos-latest]
+      fail-fast: false
+    steps:
+    - uses: ros-tooling/setup-ros2@0.0.12
+    - uses: ros-tooling/action-ros-ci@0.0.13
+      with:
+        package-name: nodl_python
+    - uses: actions/upload-artifact@master
+      with:
+        name: colcon-logs
+        path: ros_ws/log
+      if: always()
```

Comment:
From the docs:

> You can use the if conditional to prevent a job from running unless a condition is met. You can use any supported context and expression to create a conditional.

In that context, it sounds like `if: always()` would force a job to run even if previous jobs failed (as opposed to `if: success()`). Given that we don't have a previous job, do we even need this?

Reply:
When a step within a job fails, future steps fail as well. `if: always()` forces an attempt to upload the colcon logs if the previous step fails.

Reply:
Oh interesting, I missed the fact that this was part of the `step` object as opposed to the `job`. The [docs](https://help.github.com/en/actions/automating-your-workflow-with-github-actions/workflow-syntax-for-github-actions#jobsjob_idif) only show `jobs.<job_id>.if` as an option, I don't see step-level `if`'s documented. Can you point me in the right direction?

Reply:
Ah, I'm blind, [here it is](https://help.github.com/en/actions/automating-your-workflow-with-github-actions/workflow-syntax-for-github-actions#jobsjob_idstepsif). Okay, I'm on the same page now.